meeting-notes/2019-Jun-06.md

```diff
@@ -0,0 +1,47 @@
+- **Kaizhe:** I’m more looking into this to enrich how this new architecutre can bring the detection capabilities to the next Level, I see some flexibility here when we have a broker and it can’t be easily enrich the Falco events but I think I still agree with Mark, maybe start with simple and see what we can do from there
+- **Thomas:** It’s a little bit complicated to have a mechanism that will work everywhere
+** Mattia:** I don’t have a complete understanding and view on this proble, just started to learn falco last week.
+
+[#6](https://github.com/falcosecurity/office-hours/issues/6): Github and contribution automation:
+
+- **Leonardo:** we have an idea to improve the way internal and external contribute to falco, this idea is to implement and automate a lot the workflow, to make everything more structured in a way that in every moment we have statistics about the number of issues, the kind of issues, the area the issue refers to, and a set of commands, and a set of well defined rules so that for example no one can just merge their PR without following the rules. 
+The first phase is to apply this set of rules only on issues and PRs, next we will automate things like: the kind of git history we want to enforce, allowing or disallowing merge commits for the PR; implement branch protection, let’s say that we use a git flow model with feature/*, hotfix/*, master and develop branches and we want to disallow pushing against master. The tool we are experimenting with is Prow, the same tool that every CNCF project like k8s, knative, istio, prometheus, jetstack, azure k8s engine, and some other use. We can use it for all the repository in the Falco organization since it’s multi-tenant. The setup is already visible in https://github.com/falcosecurity/test-infra you can find all the yaml engineering needed there (https://yaml.engineering lulz).  We have deployed it on an EKS Kubernetes. We have a config file and a plugin file, the config file is used to define the general configuration about the plugins. We have PR sizes, reviewer auto assign, approve and looks-good-to-me chatops, among the others. Cannot explain in detail every plugin here. But I hope the general sense of such automation has been given.
+Also I want to say that in a third phase with such tool we could implement also Continuous Integration based on Kubernetes.
+- **Lorenzo:** Leo, can you show some PRs or issues automated with it?
+- **Leonardo:** Yes, and he showed closes PR on test-infra with initial automation in place, OWNERS with automatic approvers and reviewers mechanism, TIDE tool for automatic and periodic merging of PR respecting all of the conditions required to be merged, etc.
+- **Mattia:** it could be a good way to improve the workflow of the falco development and can be helpful to manage and optimize the management of the software engineering branch of falco so I guess that if there is a big project such as falco a tool like this can be very helpful.
```

Comment:
```suggestion
- **Mattia**: it could be a good way to improve the workflow of the falco development and can be helpful to manage and optimize the management of the software engineering branch of falco so I guess that if there is a big project such as falco a tool like this can be very helpful.
```